src/main/webapp/app/core/auth/account.service.ts

```diff
@@ -328,23 +329,38 @@ export class AccountService implements IAccountService {
     /**
      * Sends the added SSH key to the server
      *
-     * @param sshPublicKey
+     * @param userSshPublicKey The userSshPublicKey DTO containing the details for the new key which should be created
      */
-    addSshPublicKey(sshPublicKey: string): Observable<void> {
-        if (this.userIdentity) {
-            this.userIdentity.sshPublicKey = sshPublicKey;
-        }
-        return this.http.put<void>('api/account/ssh-public-key', sshPublicKey);
+    addNewSshPublicKey(userSshPublicKey: UserSshPublicKey): Observable<HttpResponse<UserSshPublicKey>> {
```

Comment:
as this is decoupled from the user / account in the database now, would it make sense to put it into a separate service?